ssl/ssl_lib.c

```diff
@@ -3189,6 +3190,9 @@ SSL_CTX *SSL_set_SSL_CTX(SSL *ssl, SSL_CTX *ctx)
             ssl->cert->ciphers_rawlen = ocert->ciphers_rawlen;
             ocert->ciphers_raw = NULL;
         }
+        for (i = 0; i < SSL_PKEY_NUM; i++) {
+            ssl->cert->pkeys[i].digest = ocert->pkeys[i].digest;
```

Comment:
I'm left wondering what should be happening to `pkeys[i].valid_flags`

Reply:
`valid_flags` is a function of the certificate, so if they've been computed at this point, you have worse problems. (Though yeah the real problem is you all should not be storing handshake state on `CERT`.)

Looks like they get set fairly late, when you go to pick the cipher suite.

Reply:
Friendly ping.

(Since you are all doing a release soon, it seems good to get this in for that. Having OpenSSL servers out there which only sign SHA-1 is quite problematic for the ecosystem.)

Reply:
(Storing handshake state on CERT is a historical accident caused by all other structs being public.)